### DISCUSSION TITLE
/stack

### DISCUSSION THREAD

**Discussion by MrMcyeet** [Zrips/CMI#3900] [Feature Requests]
**Description of feature/improvement request:

/stack would let you get oversized stacks of items. For instance, if you had two stacks of 16 epearls, /stack would make it one stack of 32 epearls. Or if you had 4 white beds, and did /stack, it would put them all into the same stack.
**

---
### Detailed explanation of what it should do, when and in what situation: 

It can be used any time a player has, for instance, 20 lava buckets, and they do /stack, it would put them all into the same (20 bucket) stack
###

### Basic explanation as to why this feature is needed and/or useful to have: 

Useful for allowing players to transport larger amounts of items, like beds or buckets
###

**Comment by CloudeLecaw** [comment]
you can create more items allready with more items.
a stack? really? Dont you think there is a reason for not stacking over the vanilla default? ;)
why you need a massive stacksize of these?

> more (playerName) (-clone/[amount])
> Fills item stack to maximum amount
> 
> Permissions:
> cmi.command.more.oversize  to get oversized stacks

**Comment by MrMcyeet** [comment]
I had looked into the /more command, and as far as I can tell, it only maxes out the stack of what you're holding.

My idea of a /stack would only use what you already have in your inventory, so if a player had one white bed in his inventory, held it in his and, and did /stack nothing would happen.

But If  a player had 6 white beds in his inventory, and then did /stack with a white bed in his hand, it would take all of the white beds from his inv and put them into his hand as one 6 stack of white beds.

I don't think /more can do that though.

**Comment by CloudeLecaw** [comment]
but ok since this is a suggestion...
But i have a problem with this...
each day ppl are wishing/suggresting new command for CMI.
Why should not work with existing stuff that works in a simliar cathegory instead of adding new commands all time to cmi
for small things like like... Its not usefull if CMI has 1503435 commands in 1-2 years.....

**_So what about a new varriable for this more command....? like varriable  -> (-stack)_**

**command:**
more (playerName) (-clone/[amount]/**_-stack_**)

**Permission:**
 **_cmi.command.more.oversize.stack_** -  allows using the stacking argument to stack items to the maximum of 64

**What the new varriable should do:**
- stack existing items from inventoriy to the stacksize to allowed of maximum 64
.
.
Note you should not create bigger stack then 64! this the limitation if minecraft and this also has reasons. true - you can higher the amount with plugins but this can be result in buggy stacks. so pls limit it to 64.
.
.
if this is added you can create a custom alias.
```
  stack:
    Cmds:
    - cmi more -stack
```

then your players could use the wished stack command :-)

**Comment by MrMcyeet** [comment]
The problem I have with that idea is that it's still cheat-y.
The /stack would only use what the player already had in their inventory, and thus not give them anything they didn't get on their own. 

And I agree, there is no need to make stacks of more than 64.
The idea for /stack is just for things that don't already stack to 64 (i.e. beds, buckets, ender pearls, eggs, etc...)


[EDIT: I just re-read your comment, and adding a -stack flag, (that would ONLY use what was already in the players inventory) to /more would be a great solution.  I misread what you put there]
